### PR TITLE
Comments list performance optimization

### LINKED ISSUE #5297: Page Crash on new issue
- Gitea version (or commit ref): v464dcd1
- Git version: git version 2.16.2.windows.1
- Operating system: Windows server 2012
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (https://try.gitea.io/emejia/Test_Incident/issues/new)
  - [ ] No
  - [ ] Not relevant
- Log gist: Unable to obtain One

## Description
Server crashes when you perform the following tasks in the given order :
- Browse a repo as Collaborator with write permissions
- Create new Issue  + Set Title + Set Description 
- Upload any Picture
- Set Asignee
- Set Branch
- Hit "Create Issue" button

## Screenshots
It just kills the page
sin titulo

### PR DESCRIPTION
**Pull request by lunny**
Will also fix #5297

### CODE REVIEW

**Review by zeripath — CHANGES_REQUESTED** [review]

File: models/issue_comment.go [hunk 1]
```diff
@@ -150,6 +150,35 @@ func (c *Comment) LoadIssue() (err error) {
...
+			c.PosterID = -1
+			c.Poster = NewGhostUser()
+		} else {
+			log.Error(3, "getUserByID[%d]: %v", c.ID, err)
+		}
+	}
+	return err
+}
+
+func (c *Comment) loadAttachements(e Engine) (err error) {
```
[the remark this answers is not in the document]
Comment (lunny) [reply models/issue_comment.go]: done.

File: models/issue_comment_list.go [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+		reviewIDs = reviewIDs[limit:]
+	}
+
+	for _, comment := range comments {
+		comment.Review = reviews[comment.ReviewID]
+	}
+	return nil
+}
+
+// loadAttributes loads all attributes, expect for attachments and comments
```
Comment (zeripath) [inline models/issue_comment_list.go]: s/expect/except

**Comment by zeripath** [comment]
How does this work for issues with a large number of comments?

It looks like this is going to get everything in one go.

**Comment by lunny** [comment]
@zeripath This should have better performance than before when the issue has a large number of comments. The PR will reduce many database queries when loading comments attributes.

**Comment by lafriks** [comment]
@lunny can you fix remainig issue with 500 error (see drone tests)

**Comment by lunny** [comment]
@lafriks @JonasFranzDEV will fix this.

**Comment by lunny** [comment]
@lafriks @jonasfranz fixed.

**Comment by lunny** [comment]
It's ready to review.

File: models/issue_comment_list.go [hunk 3]
```diff
@@ -56,3 +51,488 @@ func (comments CommentList) loadPosters(e Engine) error {
...
+	return keysInt64(ids)
+}
+
+// Issues returns all the issues of comments
+func (comments CommentList) Issues() IssueList {
+	var issues = make(map[int64]*Issue, len(comments))
+	for _, comment := range comments {
+		if comment.Issue != nil {
+			_, ok := issues[comment.Issue.ID]
+			if !ok {
```
Comment (lafriks) [inline models/issue_comment_list.go]: nit: This should be like in other functions:
```
if _, ok := issues[comment.Issue.ID]; !ok {
    issues[comment.Issue.ID] = comment.Issue
}
```

**Comment by lunny** [comment]
@lafriks done.
@zeripath please review.

**Review by zeripath — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**